tutorials/20_REST_API.ipynb

```diff
@@ -0,0 +1,485 @@
+    "  image: \"docker.elastic.co/elasticsearch/elasticsearch:7.9.2\"\n",
+    "...\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### New document search pipeline\n",
+    "\n",
+    "We are going to create a document search pipeline. Create a new file named `document-search.haystack-pipeline.yml` in `/pipeline` folder. Do not forget to update `PIPELINE_YAML_PATH` in `config.py` with the new file name.\n",
+    "\n",
+    "```python\n",
+    "PIPELINE_YAML_PATH = os.getenv(\n",
+    "    \"PIPELINE_YAML_PATH\", str((Path(__file__).parent / \"pipeline\" / \"document-search.haystack-pipeline.yml\").absolute())\n",
+    ")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you are using Docker Compose for Haystack API, there are further changes you need to make in `docker-compose.yml`:\n",
+    "1. Create `volumes` with the path to the `document-search.haystack-pipeline.yml` file:\n",
+    "    ```yaml\n",
+    "    ...\n",
+    "      haystack-api:\n",
+    "        image: \"deepset/haystack:cpu-main\"\n",
+    "        volumes:\n",
+    "          - /<path_to_haystack>/haystack/rest_api/rest_api/pipeline:/home/user/rest_api/pipeline\n",
+    "    ...\n",
+    "    ``` \n",
+    "2. Update the `PIPELINE_YAML_PATH` value as `/home/user/rest_api/pipeline/document-search.haystack-pipeline.yml`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This document search pipeline only requires a Retriever. Therefore, it is going to be enough for our query pipeline if we declare a DocumentStore and Retriever in `document-search.haystack-pipeline.yml`. \n",
+    "\n",
+    "Learn more about YAML files in [YAML File Definitions](https://docs.haystack.deepset.ai/docs/pipelines#yaml-file-definitions). \n",
+    "\n",
+    "```yaml\n",
+    "...\n",
+    "components:\n",
+    "  - name: DocumentStore\n",
+    "    type: ElasticsearchDocumentStore\n",
+    "    params:\n",
+    "      host: localhost\n",
+    "      embedding_dim: 768\n",
+    "  - name: Retriever\n",
+    "    type: EmbeddingRetriever\n",
+    "    params:\n",
+    "      document_store: DocumentStore\n",
+    "      embedding_model: sentence-transformers/multi-qa-mpnet-base-dot-v1 \n",
+    "      model_format: sentence_transformers\n",
+    "      top_k: 5 \n",
+    "...\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And the query pipeline with a Retriever should look like this:\n",
+    "\n",
+    "```yaml\n",
+    "...\n",
+    "pipelines:\n",
+    "  - name: query\n",
+    "    nodes:\n",
+    "      - name: Retriever\n",
+    "        inputs: [Query]\n",
+    "...\n",
+    "```\n",
+    "Be sure to have the same `name` with `QUERY_PIPELINE_NAME` variable in `config.py`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Indexing pipeline\n",
```

Comment:
If we are going to suggest performing indexing via the /file-upload endpoint, I think we need to give a bit more guidance as to what this actually entails. What does the curl command look like?